node/tests/schemaStruct.go

```diff
@@ -101,16 +102,32 @@ func SchemaTestStructNesting(t *testing.T, engine Engine) {
 		t.Run("typed-read", func(t *testing.T) {
 			Require(t, n.Kind(), ShouldEqual, ipld.Kind_Map)
 			Wish(t, n.Length(), ShouldEqual, int64(1))
+
 			n2 := must.Node(n.LookupByString("x"))
 			Require(t, n2.Kind(), ShouldEqual, ipld.Kind_Map)
+
+			n2Seg := must.Node(n.LookupBySegment(ipld.PathSegmentOfString("x")))
+			Wish(t, ipld.DeepEqual(n2, n2Seg), ShouldEqual, true)
+
+			n2Node := must.Node(n.LookupByNode(basicnode.NewString("x")))
+			Wish(t, ipld.DeepEqual(n2, n2Node), ShouldEqual, true)
```

Comment:
I'm not sure if this is a hair worth splitting or not, but... I think you could actually use pointer equality here for these last two equality checks (or native equality in general).  Thoughts?

Reply:
I would say not, because that still assumes one can use == on Nodes, which might not be true. It's an interface, after all.